packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3458,3 +3462,66 @@ const waitForSeq = async (
 			deltaManager.on("op", handleOp);
 		}
 	});
+
+/**
+ * Mixin class that adds await for DataObject to finish initialization before we proceed to summary.
+ * @param handler - handler that returns info about blob to be added to summary.
+ * Or undefined not to add anything to summary.
+ * @param Base - base class, inherits from FluidDataStoreRuntime
+ */
+export const mixinSummaryHandler = (
+	handler: (
+		runtime: ContainerRuntime,
+	) => Promise<{ path: string[]; content: string } | undefined>,
+	Base: typeof ContainerRuntime = ContainerRuntime,
+) =>
+	class RuntimeWithSummarizerHandler extends Base {
+		async summarize(options) {
+			const result = await super.summarize(options);
+			const content = await handler(this);
+			if (content !== undefined) {
+				// all user added blobs wil be in ".application" subtree
+				content.path.unshift(".application");
+				addBlobToSummaryViaPath(
+					result.summary,
+					result.stats,
+					content.path,
+					content.content,
+				);
+			}
+			return result;
+		}
+	} as typeof ContainerRuntime;
+
+// Helper API to load extra blobs in snapshot
+// This API only works for blobs that are saved in storage, i.e. this API would only work for load from storage scenarios
+export async function loadRuntimeBlob(
```

Comment:
What is this for? Looks like its only used in a test and should move to that test. It does not look like something we would like to be part of the public API.

Reply:
i think this is for loading a blob immediately after the containerRuntime is booted. for loop to faster load a blob they want